crates/fluvio-spu/src/kv/consumer.rs

```diff
@@ -0,0 +1,295 @@
+use std::{
+    time::{Duration, SystemTime},
+    sync::Arc,
+    collections::{HashMap, hash_map::Entry},
+    ops::AddAssign,
+};
+
+use anyhow::Result;
+use async_rwlock::RwLock;
+use fluvio_kv_storage::KVStorage;
+use fluvio_protocol::{record::ReplicaKey, Encoder, Decoder};
+use fluvio_storage::FileReplica;
+use fluvio_types::PartitionId;
+use tracing::trace;
```

Comment:
can separate out third party and fluvio dep?

Reply:
Sure, done